plugin/klogrus/klogrus.go

```diff
@@ -0,0 +1,63 @@
+package klogrus
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+// Logger provides the kgo.Logger interface for usage in kgo.WithLogger when
+// initializing a client.
+type Logger struct {
+	lr *logrus.Logger
+}
+
+// New returns a new Logger.
+func New(lr *logrus.Logger) *Logger {
+	return &Logger{lr}
+}
+
+// Level is for the kgo.Logger interface.
+func (l *Logger) Level() kgo.LogLevel {
+	return logrusToKgoLevel(l.lr.GetLevel())
+}
+
+// Log is for the kgo.Logger interface.
+func (l *Logger) Log(level kgo.LogLevel, msg string, keyvals ...any) {
+	logrusLevel, levelMatched := kgoToLogrusLevel(level)
+	if levelMatched {
+		fields := make(logrus.Fields, len(keyvals)/2)
+		for i := 0; i < len(keyvals); i += 2 {
+			fields[keyvals[i].(string)] = keyvals[i+1]
```

Comment:
I'm going to change this to `fmt.Sprint(keyvals[i])` in a last-minute PR just for safety